src/App.tsx

```diff
@@ -39,6 +42,28 @@ function App() {
 
   // function seekSong()
 
+  useEffect(() => {
+    console.log("apiservices changed");
+    if (apiServices) {
+      console.log(apiServices, "apiServices");
+      console.log(apiServices.getTop);
+      apiServices
+        .getTop()
+        .then((res) => {
+          if (res.status === 401) {
+            window.location.href = import.meta.env.VITE_REDIRECT_URI;
+          }
+        })
+        .catch((error) => console.log("error", error));
+      // .catch((error) => {
+      //   console.log("error", error);
+      //   // window.location.href = import.meta.env.VITE_REDIRECT_URI;
+      // });
+      // setTopItems(apiServices.getTop());
+      // apiServices.getMe().then((res) => console.log("getme", res));
+    }
+  }, [apiServices]);
+
   useEffect(() => {
     setApiServices(SpotifyServices(accessToken));
```

Comment:
is there any chance we only want to do this conditionally? or do we want to reset the ApiServices every time the accessToken updates?

Reply:
maybe just leave a comment asserting this and why:

"every time the accessToken is updated, we want to setApiServices to the SpotifyServices corresponding with that accessToken. We want to do this because XYZ."